app/PickMe_nebula0/app/src/main/java/com/example/pickme_nebula0/notification/MessageViewActivity.java

```diff
@@ -82,19 +88,29 @@ private void loadMessages(){
                 .addOnCompleteListener(task -> {
                     if (task.isSuccessful()) {
                         for (QueryDocumentSnapshot document : task.getResult()) {
+                            // get a notification
                             Notification n = document.toObject(Notification.class);
 
-                            notifs.add(n);
+                            dbManager.notificationShouldExist(
+                                    n,
+                                    () -> {
+                                        // if it should exist, add it
+                                        notifs.add(n);
+                                        notifAdapter.notifyDataSetChanged();
+                                        // show notifications from newest to oldest
+                                        notifs.sort(new Comparator<Notification>() {
+                                            @Override
+                                            public int compare(Notification n1, Notification n2) {
+                                                return n2.getTimestamp().compareTo(n1.getTimestamp());
+                                            }
+                                        });
```

Comment:
- we don't need to notify the adapter of changes to the list until we have finished updating it (e.g. outside the for loop)

- also, it's critical that we sort the list before we tell the adapter that we've updated it.
with the current implementation, the very last notification to be added may not be in the right order

Reply:
1. When `notifyDataSetChanged()` is moved after the `for loop`, none of the messages show up in the `listview`. I couldn't figure it out, so I notified the adapter every time a notification was added.
2. Moved the `notifyDataSetChanged()` after `sort`.